### PR TITLE
[DOCS] Add `number_of_routing_shards` index setting to index modules

### LINKED ISSUE #60182: Explain index.number_of_routing_shards for splitting
This setting is mentioned in the [split docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-split-index.html#), but nowhere else that I can find. It feels like it should be mentioned [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-modules-settings), or perhaps given a practical example in the split docs?

One thing that's worth adding as well, is that you cannot change `number_of_routing_shards` on an open index.

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Adds the `number_of_routing_shards` index setting to index modules docs.
* Updates the split API docs to mention that `number_of_routing_shards`
is a static setting.

Closes #60182

### CODE REVIEW

File: docs/reference/index-modules.asciidoc [hunk 1]
```diff
@@ -46,6 +46,17 @@ NOTE: The number of shards are limited to `1024` per index. This limitation is a
 
 // end::index-number-of-shards-tag[]
 
+[[index-number-of-routing-shards]]
+`index.number_of_routing_shards`::
+Number of routing shards used to <<indices-split-index,split>> an index and its
```
Comment (andreidan) [inline docs/reference/index-modules.asciidoc:51]: I think the "split an index and its primary shards" sounds a bit confusing. Would ending the sentence after `split an index` makes more sense?

File: docs/reference/index-modules.asciidoc [hunk 2]
```diff
@@ -46,6 +46,17 @@ NOTE: The number of shards are limited to `1024` per index. This limitation is a
...
+primary shards. This setting specifies the hashing space used internally to
+distribute documents for a split index across shards. See
+<<indices-split-index>>.
```
Comment (andreidan) [inline docs/reference/index-modules.asciidoc:54]: I think this is too abstract. Shall we include the example from the split-index documentation here as well? It'd be a bit of duplication but I don't know if this is clear enough as to how it works without an example

**Review by andreidan — APPROVED** [review]
Thanks for working on this James. 

I left a couple of suggestions (I find it a very tricky setting)

**Comment by jrodewig** [comment]
Thanks @andreidan

**Pull request merged**